liquidctl/driver/asetek_pro.py

```diff
@@ -0,0 +1,271 @@
+
+from liquidctl.driver.asetek import Modern690Lc
+from liquidctl.driver.usb import UsbDriver
+from liquidctl.error import NotSupportedByDevice
+from liquidctl.keyval import RuntimeStorage
+from liquidctl.util import clamp
+
+
+LOGGER = logging.getLogger(__name__)
+
+_MAX_PROFILE_POINTS = 7
+_CRITICAL_TEMPERATURE = 60
+_HIGH_TEMPERATURE = 45
+_READ_LENGTH = 32
+_READ_TIMEOUT = 2000
+_WRITE_TIMEOUT = 2000
+
+_PRO_WRITE_ENDPOINT = 0x1
+_PRO_READ_ENDPOINT = 0x81
+_PRO_CMD_READ_FIRMWARE = 0xaa
+_PRO_CMD_READ_AIO_TEMP = 0xa9
+_PRO_CMD_READ_FAN_SPEED = 0x41
+_PRO_CMD_WRITE_FAN_SPEED = 0x40
+_PRO_CMD_READ_PUMP_SPEED = 0x31
+_PRO_CMD_READ_PUMP_MODE = 0x33
+_PRO_CMD_WRITE_PUMP_MODE = 0x32
+_PRO_PUMP_MODES = [
+    'Quiet',
+    'Balanced',
+    'Performance'
+    ]
+
+_PRO_SPEEDS = {
+    'rainbow': [
+        0x30, # slow
+        0x18, # medium
+        0x0C  # fast
+    ],
+    'shift': [
+        0x46, # slow
+        0x28, # medium
+        0x0F  # fast
+    ],
+    'pulse':[
+        0x50, # slow
+        0x37, # medium
+        0x1E  # fast
+    ],
+    'blinking': [
+        0x0F, # slow
+        0x0A, # medium
+        0x05  # fast
+    ]
+}
+
+
+class CorsairAsetekProDriver(Modern690Lc):
+    """liquidctl driver for Corsair-branded sixth generation Asetek coolers."""
+    SUPPORTED_DEVICES = [
+        (0x1b1c, 0x0c12, None, 'Corsair Hydro H150i Pro', {}),
+        (0x1b1c, 0x0c13, None, 'Corsair Hydro H115i Pro', {}),
+        (0x1b1c, 0x0c14, None, 'Corsair Hydro H110i Pro', {}),
+        (0x1b1c, 0x0c15, None, 'Corsair Hydro H100i Pro', {}),
+        (0x1b1c, 0x0c16, None, 'Corsair Hydro H80i Pro', {}),
+    ]
+
+    def _write(self, data):
+        """Write data to the AIO and log"""
+        LOGGER.debug('write %s', ' '.join(format(i, '02x') for i in data))
+        self.device.write(_PRO_WRITE_ENDPOINT, data, _WRITE_TIMEOUT)
+
+    # Not calling this (or begin transaction) since they do not seem to be needed for H100i Pro
+    def _end_transaction_and_read(self, length=_READ_LENGTH):
+        """End the transaction by reading from the device.
+        According to the official documentation, as well as Craig's open-source
+        implementation (libSiUSBXp), it should be necessary to check the queue
+        size and read data in chunks.  However, leviathan and its derivatives
+        seem to work fine without this complexity; we currently try the same
+        approach.
+        """
+        msg = self.device.read(_PRO_READ_ENDPOINT, length, _READ_TIMEOUT)
+        LOGGER.debug('received %s', ' '.join(format(i, '02x') for i in msg))
+        self.device.release()
+        return msg
+
+    def _write_color_change_speed(self, mode, speed):
+        """Send the speed to cycle colors on the RGB pump"""
+        if speed < 1 or speed > 3:
+            raise ValueError('Unsupported speed ({}), only 1-3 supported'.format(speed))
+
+        self._write([0x53, _PRO_SPEEDS[mode][int(speed)-1]])
+        self._end_transaction_and_read(3)
```

Comment:
If this is unused code can it be removed?

Reply:
This function does seem to be called, maybe the comment is just out of date?

Reply:
This code should be called from the `set_color` function.